File: src/vanilla/DanModel/JavaBlock.cs
using System;

namespace AutoRest.Java.DanModel
{
    public class JavaBlock
    {
        private readonly JavaFileContents contents;

        public JavaBlock(JavaFileContents contents)
        {
            this.contents = contents;
        }

        public JavaBlock Indent(Action indentAction)
        {
            contents.Indent(indentAction);
            return this;
        }

        public JavaBlock IncreaseIndent()
        {
            contents.IncreaseIndent();
            return this;
        }

        public JavaBlock DecreaseIndent()
        {
            contents.DecreaseIndent();
            return this;
        }

        public JavaBlock Line(string text, params object[] formattedArguments)
        {
            contents.Line(text, formattedArguments);
            return this;
        }

        public JavaBlock Line()
        {
            contents.Line();
            return this;
        }

        public JavaBlock Block(string text, Action<JavaBlock> bodyAction)
        {
            contents.Block(text, bodyAction);
            return this;
        }

        public JavaBlock SingleLineComment(string text)
        {
            contents.SingleLineComment(text);
            return this;
        }

        public void MultipleLineComment(Action<JavaMultipleLineComment> commentAction)
        {
            contents.MultipleLineComment(commentAction);
        }

        public JavaBlock Return(string text)
        {
            contents.Return(text);
            return this;
        }

        public void Annotation(params string[] annotations)
        {
            contents.Annotation(annotations);
        }

        public void ReturnAnonymousClass(string anonymousClassDeclaration, Action<JavaClass> anonymousClassBlock)
        {
            contents.ReturnAnonymousClass(anonymousClassDeclaration, anonymousClassBlock);
        }

        public JavaIfBlock<JavaBlock> If(string condition, Action<JavaBlock> ifAction)
        {
            contents.If(condition, ifAction);
            return new JavaIfBlock<JavaBlock>(this, contents);
        }
    }
}

File: mapper/src/main/java/com/azure/autorest/model/javamodel/JavaBlock.java
package com.azure.autorest.model.javamodel;

import java.util.function.Consumer;

public class JavaBlock implements JavaContext
{
    private JavaFileContents contents;

    public JavaBlock(JavaFileContents contents)
    {
        this.contents = contents;
    }

    public final void Indent(Runnable indentAction)
    {
        contents.Indent(indentAction);
    }

    public final void IncreaseIndent()
    {
        contents.IncreaseIndent();
    }

    public final void DecreaseIndent()
    {
        contents.DecreaseIndent();
    }

    public final void Text(String text)
    {
        contents.Text(text);
    }

    public final void Line(String text, Object... formattedArguments)
    {
        contents.Line(text, formattedArguments);
    }

    public final void Line()
    {
        contents.Line();
    }

    public final void Block(String text, Consumer<JavaBlock> bodyAction)
    {
        contents.Block(text, bodyAction);
    }

    public final void JavadocComment(String text)
    {
        contents.JavadocComment(text);
    }

    public final void JavadocComment(Consumer<JavaJavadocComment> commentAction)
    {
        contents.JavadocComment(commentAction);
    }

    public final void Return(String text)
    {
        contents.Return(text);
    }

    public final void Annotation(String... annotations)
    {
        contents.Annotation(annotations);
    }

    public final void ReturnAnonymousClass(String anonymousClassDeclaration, Consumer<JavaClass> anonymousClassBlock)
    {
        contents.ReturnAnonymousClass(anonymousClassDeclaration, anonymousClassBlock);
    }

    public final JavaIfBlock If(String condition, Consumer<JavaBlock> ifAction)
    {
        contents.If(condition, ifAction);
        return new JavaIfBlock(contents);
    }

    public final void Lambda(String parameterType, String parameterName, Consumer<JavaLambda> body)
    {
        contents.Lambda(parameterType, parameterName, body);
    }

    public final void Lambda(String parameterType, String parameterName, String returnExpression)
    {
        contents.Lambda(parameterType, parameterName, returnExpression);
    }
}